src/sanitation.js

```diff
@@ -241,16 +250,26 @@ export function isValidAttr(tagName, attrName, attrValue, opt_purify = false) {
   }
 
   // Remove blacklisted attributes from specific tags e.g. input[formaction].
-  const attrNameBlacklist = BLACKLISTED_TAG_SPECIFIC_ATTRS[tagName];
-  if (attrNameBlacklist && attrNameBlacklist.indexOf(attrName) != -1) {
+  let attrBlacklist = BLACKLISTED_TAG_SPECIFIC_ATTRS[tagName] || [];
+  if (isAmp4Email(doc)) {
+    const amp4EmailAttrBlacklist =
+        {'form': ['name'], 'amp-anim': ['controls']}[tagName] || [];
+    attrBlacklist = attrBlacklist.concat(amp4EmailAttrBlacklist);
+  }
+  if (attrBlacklist.indexOf(attrName) != -1) {
     return false;
   }
 
   // Remove blacklisted values for specific attributes for specific tags
   // e.g. input[type=image].
-  const attrBlacklist = BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES[tagName];
-  if (attrBlacklist) {
-    const blacklistedValuesRegex = attrBlacklist[attrName];
+  let attrValueBlacklist =
+      BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES[tagName];
+  if (isAmp4Email(doc)) {
+    attrValueBlacklist =
+        EMAIL_BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES[tagName];
```

Comment:
Why do we overwrite here but concat/merge above (line 257)?

Reply:
Fixed to be consistent with the overriding here which you said was consistent with how validator did things, we can just not concat and duplicate the values from BLACKLISTED_TAG_SPECIFIC_ATTRS into a new blacklist specific to AMP4Email.